// data-analysis-python\src\get_stock_data_with_dates\get_stock_data_with_dates.proto
syntax = "proto3";  // プロトコルバッファのバージョンを指定

option go_package = "api-go/src/service/ms_gateway/get_stock_data_with_dates";

// GetStockServiceというサービスを定義
service GetStockDataWithDatesService {
  rpc GetStockData (GetStockDataWithDatesRequest) returns (GetStockDataWithDatesResponse);
}

// リクエストメッセージを定義
message GetStockDataWithDatesRequest {
  string ticker = 1;  // 銘柄コードを格納するフィールド
  string start_date = 2;  // 期間を格納するフィールド
  string end_date = 3;  // 期間を格納するフィールド
}

// 株価データを持つメッセージを定義
message StockDataWithDates {
  double open = 1;
  double close = 2;
  double high = 3;
  double low = 4;
  double volume = 5;
}

// レスポンスメッセージを定義
message GetStockDataWithDatesResponse {
  map<string, StockDataWithDates> stock_data = 1;  // 期間分の株価データを含むフィールド  -> map<日付, 株価データ>
  string stock_name = 2; // 銘柄名を含むフィールド
}



// python用コンパイルコマンド /data-analysis-pythonにて
// python -m grpc_tools.protoc -I src/get_stock_data_with_dates --python_out src/get_stock_data_with_dates --grpc_python_out=src/get_stock_data_with_dates src/get_stock_data_with_dates/get_stock_data_with_dates.proto

// Go用コンパイルコマンド /api-goにて
// protoc --proto_path=../data-analysis-python/src/get_stock_data_with_dates --go_out=./src/service/ms_gateway/get_stock_data_with_dates --go_opt=paths=source_relative --go-grpc_out=./src/service/ms_gateway/get_stock_data_with_dates --go-grpc_opt=paths=source_relative ../data-analysis-python/src/get_stock_data_with_dates/get_stock_data_with_dates.proto